libraries/AP_Param/AP_Param.h

```diff
@@ -805,6 +824,7 @@ class AP_ParamT : public AP_Param
     ///
     void set_default(const T &v) {
         if (!configured()) {
+            add_default(this, (float)v);
             set(v);
         }
```

Comment:
I think from what @tridge said about his version this should be:
```suggestion
        add_default(this, (float)v);
        if (!configured()) {
            set(v);
        }
```
So we always get the correct default value no matter if the param is setup or not.